### DISCUSSION TITLE
Syncing timers across devices?

### DISCUSSION THREAD

**Discussion by ankushg** [leikoilja/ha-google-home#446] [Q&A]
One of my biggest annoyances with Google Home devices is that timers aren't synced across devices.

Example:
- I'm cooking in the kitchen, put some stuff in the oven, and set a timer on my kitchen Nest Hub for 40 minutes
- I go to the living room
- I ask the living room Nest Mini how much time is left on the timer: it tells me that there is no timer!
- I have to go back to the kitchen to ask the kitchen Nest Hub what the timer situation is

This gets even more frustrating when more than one device is within earshot: it's hard to tell which device even set up the timer!

---

Is it possible to use this integration to automatically:
- detect when a new timer is created on one device
- automatically "clone" it on all other devices

and

- detect when a timer is stopped on one device
- automatically stop the equivalent timer on all other devices?

**Comment by DurgNomis-drol** [comment]
In short, no. The API we uses does not have support for this

**Reply by leikoilja** [reply]
@DurgNomis-drol is right. Unfortunately, Google API that was reverse engineered and we use does not support to set the new alarms or timers, we can only read or delete them.

What you could technically do to play around with automations is to:
1. Set a timer on a Google Home
2. Read that timer on Home Assistant as a virtual timer sensor
3. Remove the timer from you Google Home
4. Have a virtual timer sensor 
5. Make HA to play a tune on all your music devices when the virtual timer goes off

And to stop that tune you can create another HA sensor like 'playing' which will just be a toggle.
When the tune starts playing your automation sets the sensor to 'on' and when you want to stop it you just tell your Google Home to 'turn off <name of your sensor' which should make HA to stop the tune from playing.

A bit of a workaround and a hacky thing to do, but a fun project to try :D 
Let us know if you ll find any working solution :)

**Reply by KapJI** [reply]
That's pretty clever solution 😀
For running some tunes on group of speakers you can use [Spotcast](https://github.com/fondberg/spotcast) integration.
